applications/mne_analyze/plugins/channelselection/channelselection.cpp

```diff
@@ -263,3 +267,12 @@ void ChannelSelection::onSelectionChanged(const QList<QGraphicsItem*>& selectedC
 
     m_pCommu->publishEvent(EVENT_TYPE::CHANNEL_SELECTION_ITEMS, QVariant::fromValue(/*static_cast<void*>(*/m_pSelectionItem/*)*/));
 }
+
+//=============================================================================================================
+
+void ChannelSelection::onModelRemoved(QSharedPointer<ANSHAREDLIB::AbstractModel> pRemovedModel)
+{
+    if(m_pAnalyzeData->getModelsByType(ANSHAREDLIB_FIFFRAW_MODEL).size() == 0 && m_pAnalyzeData->getModelsByType(ANSHAREDLIB_AVERAGING_MODEL).size() == 0){
+//        m_pChannelSelectionView->clearView();
```

Comment:
why is the commented out? also, what if only one of the model types is not present in analyze data? then the view would not be cleared? Not sure but maybe this should be an OR statement?

Reply:
I initially wanted to clear the view if no more models that used the channel selection (right now avg and fiffraw) were left. I then thought it would be odd to clear this view since it isn't directly related to an item being deleted in the data manager, it it would be more like clearing a settings page, so icommented out the function but forgot to remove receiving the event. Should the channel selection clear?

Reply:
ok. no, I think that is fine. thx for the explanation.